### DISCUSSION TITLE
VRR / adaptive sync in hyprland, does it work with fullscreen games?

### DISCUSSION THREAD

**Discussion by neon-grim** [hyprwm/Hyprland#919] [Q&A] [answered]
Hi all, I'm thinking of making a minimal fedora install with hyprland as my WM, since after trying sway I quite like the experience of using and customizing a tiling WM over a desktop environment like gnome or KDE.

That said, I had to revert back to Gnome 42 from sway, since [this adaptive sync bug](https://github.com/swaywm/sway/issues/6832) ruins my gaming experience. Essentially adaptive sync stops syncing with the focused game, whenever the mouse is moved regardless if its visible. In addition, the sway developer sadly isnt prioritizing this bug and thus will probably go unfixed for a long time.

Because of that, I would like to make sure that VRR works in hyprland, before I go through the trouble of setting it up. Thanks for the info in advance.

**Comment by vaxerski** [comment, marked as answer]
Well, it works, just like in Sway.

Adaptive sync works in the example you're describing. The thing is, since we are doing page flips, every commit has to repaint 2 frames. That means, if you cap to 100, your refresh rate will be 200. (Which is the case on my 240Hz display)

What you are describing is possible, but since Hyprland doesn't support direct scanout yet, it's not something on my radar *for now*. Once that is impl'd, doing this will not be a problem.

Basically, when doing a direct scanout we can clear the damage buffer and wait for the scanout'd surface to commit the next update. Without that we cannot really do that, as we don't want weird half-painted frames or flicker.

The most likely reason why emersion doesn't want to investigate this is most likely that it'd require a bit of wlroots hacking.

**Reply by neon-grim** [reply]
Thanks for the rapid reply. So from my understanding hyprland doubles the refresh rate of the application in question, in order to maintain smoother cursor movement while adhering to said applications frame pacing. So with my 160hz monitor, I'd have to target 80 fps ingame.

Sorry if I've misinterpreted the info you posted, I'm still rather new to linux (6 months). For now, I'll stay on gnome 43 until VRR works as expected on hyprland or sway. That said, thanks again for your reply.
